File: financeiro-api/src/main/java/br/com/iftm/financeiroapi/model/repository/CategoryRepository.java
package br.com.iftm.financeiroapi.model.repository;


import br.com.iftm.financeiroapi.model.domain.Category;
import org.springframework.stereotype.Component;
import org.springframework.stereotype.Repository;

import java.util.List;

@Component
public interface CategoryRepository  {

    void save(Category category);

    void update(Category category);

    void delete(String id);

    Category findById(String id);

    List<Category> findAll();

}

File: financeiro-api-kotlin/src/main/kotlin/br/com/iftm/financeiroapi/model/repository/CategoryRepository.kt
package br.com.iftm.financeiroapi.model.repository

import br.com.iftm.financeiroapi.model.domain.Category
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface CategoryRepository : JpaRepository<Category, Long>

